fairseq/training.py

```diff
@@ -0,0 +1,434 @@
+                            [
+                                "unzip",
+                                "-q",
+                                "-n",
+                                cached_audio_zip_dir,
+                                "-d",
+                                local_unzipped_dir[-1],
+                            ]
+                        )
+                    except sp.CalledProcessError:
+                        print(
+                            f"Cache manager: Error occurred for caching and unzipping audio data in {local_unzipped_dir[-1]}"
+                        )
+                        raise
+                common_audio_dir = os.path.commonpath(local_unzipped_dir)
+                for name in ["train.tsv", "valid.tsv"]:
+                    with open(os.path.join(manifest_path, name), "r") as manifest_file:
+                        manifest_lines = manifest_file.read().splitlines()
+                    for i in range(1, len(manifest_lines)):
+                        to_check = (
+                            os.path.join(common_audio_dir, manifest_lines[i].split()[0])
+                        )
+                        assert os.path.exists(
+                            to_check
+                        ), f"Manifest file {to_check} not found in unzipped directory"
+                    manifest_lines[0] = common_audio_dir
+                    with open(
+                        os.path.join(self.out_cached_audio_manifest.get_path(), name), "w"
+                    ) as cached_audio_manifest_file:
+                        cached_audio_manifest_file.write("\n".join(manifest_lines))
+
+        my_env = os.environ
+        if self.fairseq_root is not None:
+            my_env["PYTHONPATH"] = self.fairseq_root
+        sp.check_call(self._get_run_cmd(), env=my_env)
+
+    def plot(self):
+        directory = "./outputs"
+        train_score, train_error = {}, {}
+        valid_score, valid_error = {}, {}
+        learning_rates = {}
+
+        for cur in os.walk(directory):
+            dir_path = cur[0]
+            files = cur[2]
+            if "hydra_train.log" in files:
+                with open(f"{dir_path}/hydra_train.log", "r") as f:
+                    lines = f.readlines()
+                    for i in range(len(lines) - 1):
+                        line = lines[i]
+                        if 'begin validation on "valid" subset' in line:
+                            split = re.sub('[{,":]', "", lines[i + 1]).split(" ")
+                            #  Exception handling for cases in which accuracy and loss 
+                            #  are not logged in next line
+                            try:
+                                epoch = int(split[split.index("epoch") + 1])
+                                score = float(split[split.index("valid_loss") + 1])
+                                error = 1 - float(
+                                    split[split.index("valid_accuracy") + 1]
```

Comment:
for my use case,ie audio-finetuning, there is no "valid_accuracy" or "train_accuracy"

here is an example of the dev log file line:
[2022-11-02 11:38:35,221][dev][INFO] - {"epoch": 501, "dev_loss": "90.382", "dev_ntokens": "608.174", "dev_nsentences": "4.01144", "dev_nll_loss": "0.596", "dev_uer": "9.178", "dev_wer": "22.85", "dev_raw_wer": "22.85", "dev_wps": "5192.6", "dev_wpb": "608.2", "dev_bsz": "4", "dev_num_updates": "20000", "dev_best_wer": "22.85"}

and an example of the train log line:
[2022-11-02 11:39:28,709][train][INFO] - {"epoch": 501, "train_loss": "174.778", "train_ntokens": "20434.2", "train_nsentences": "125.719", "train_nll_loss": "1.075", "train_wps": "2207", "train_ups": "0.11", "train_wpb": "20434.2", "train_bsz": "125.7", "train_num_updates": "20000", "train_lr": "2.5e-06", "train_gnorm": "141.885", "train_loss_scale": "8", "train_train_wall": "137", "train_gb_free": "6.1", "train_wall": "0"}

Reply:
Well, in this case the try-except block will run into the `ValueError` case and then silently deliver empty arrays for `epoch`, `score`, and `error`. So I guess the job will work but the plots will just be empty.

We could do separate try-except blocks for loss and accuracy so you can have the loss plot even when no accuracy is reported.

Reply:
@nishthajain1611  do you have two separate log files?